server/handler/check_run.go

```diff
@@ -77,8 +77,23 @@ func (h *CheckRun) Handle(ctx context.Context, eventType, deliveryID string, pay
 		if err != nil {
 			return err
 		}
-		if err := h.ProcessPullRequest(ctx, pullCtx, client, config, fullPR); err != nil {
-			logger.Error().Err(errors.WithStack(err)).Msg("Error processing pull request")
+
+		var didUpdatePR = false
+		if h.DisableUpdateFeature {
+			logger.Debug().Msgf("Skipping updates to pull request due to server configuration override")
+		} else {
+			base, _ := pullCtx.Branches()
+			var err error
+			didUpdatePR, err = h.UpdatePullRequest(logger.WithContext(ctx), pullCtx, client, config, pr, base)
+			if err != nil {
+				logger.Error().Err(errors.WithStack(err)).Msg("Error updating pull request")
+			}
```

Comment:
I think it will be a little simpler (and match the pull request handler) to invert this condition: instead of checking `!didUpdatePR` outside of this branch, check `didUpdatePR` here and continue.
```suggestion
			didUpdatePR, err := h.UpdatePullRequest(logger.WithContext(ctx), pullCtx, client, config, pr, base)
			if err != nil {
				logger.Error().Err(errors.WithStack(err)).Msg("Error updating pull request")
			}
			if didUpdatePR {
				continue
			}
```

Reply:
Good suggestion, happy to make this change

Reply:
Addressed for both check_run and status events in https://github.com/palantir/bulldozer/pull/307/commits/92aad8fdb5aae4a22b594ab6902b883ec03908bb